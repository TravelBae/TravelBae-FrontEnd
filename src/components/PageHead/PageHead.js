import React from 'react';
import PropTypes from 'prop-types';
import Breadcrumbs from '../Breadcrumbs';
import styles from './styles.scoped.css';

export default function PageHead(props) {
  const { className, items } = props;
  const classes = [
    styles.root,
    className,
  ].filter(Boolean).join(' ');

  return (
    <header className="root">
      <Breadcrumbs items={items} />
    </header>
  );
}

PageHead.defaultProps = {
  className: '',
  items: [],
};

PageHead.propTypes = {
  className: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    url: PropTypes.string
  })),
};
