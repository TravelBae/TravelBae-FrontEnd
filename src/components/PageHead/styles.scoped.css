.root {
  background-color: var(--white-color);
  position: sticky;
  top: 0;
  
}

.root > div {
  align-items: center;
  gap: 1rem;
  display: flex;
}