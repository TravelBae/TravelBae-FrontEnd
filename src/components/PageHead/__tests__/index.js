import React from 'react';
import { useHistory } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import PageHead from '../PageHead';

describe('src/components/elements/PageHead', () => {
  test('render', () => {
    const data = { isFilter: true, isButton: true, profile: { name: 'test' } };
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PageHead {...data} />);
    PageHead.defaultProps.onClick();
    expect(tree).toMatchSnapshot();
  });

  test('without custom icon and items === 2', () => {
    const items = [
      { name: 'Dashboard', url: '/' },
      { name: 'Pendaftaran', url: '/new-user' }
    ];
    const result = PageHead({ ...PageHead.defaultProps, items, icon: '' });
    expect(result.props.children[1].props.children[0]).toBe(null);
  });

  test('without custom icon and items !== 2', () => {
    const items = [
      { name: 'Dashboard', url: '/' },
      { name: 'Pendaftaran', url: '/new-user' },
      { name: 'Tahap Pendaftaran Pelanggan' }
    ];
    const history = { push: jest.fn() };
    useHistory.mockImplementationOnce(() => history);
    const result = PageHead({ ...PageHead.defaultProps, items, icon: '' });
    result.props.children[1].props.children[0].props.onClick();
    expect(history.push).toHaveBeenCalledWith('/new-user');
  });

  test('with custom icon', () => {
    const items = [
      { name: 'Dashboard', url: '/' },
      { name: 'Pendaftaran', url: '/new-user' },
      { name: 'Tahap Pendaftaran Pelanggan' }
    ];
    const result = PageHead({ ...PageHead.defaultProps, items, icon: 'ic-back' });
    expect(result.props.children[1].props.children[0].props.src).toBe('ic-back');
  });
});
