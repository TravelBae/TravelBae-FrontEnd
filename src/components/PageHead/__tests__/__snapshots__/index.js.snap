// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/PageHead render 1`] = `
<header
  className=""
>
  <Breadcrumbs
    className=""
    items={Array []}
  />
  <div>
    <h4>
      
    </h4>
    <div>
      <Filters
        filterOptions={Array []}
        isTransactionFilter={false}
      />
      <Button
        className=""
        color="primary"
        disabled={false}
        fixed={false}
        icon={<Plus />}
        isLoading={false}
        onClick={[Function]}
        rounded={false}
        size="medium"
        to=""
        type="button"
        variant="filled"
      >
        Tambah
      </Button>
    </div>
    
  </div>
  <div>
    <div>
      <img
        alt="user"
        src="/assets/img-empty.svg"
      />
      <h4>
        test
        <br />
        <span />
      </h4>
    </div>
  </div>
</header>
`;
